### PR TITLE
lint: disallow core/dist/ imports

### PR DESCRIPTION
**Pull request by hayemaxi**
Problem: Auto-complete sometimes imports from the dist folder of the core lib, if you are working in another subproject.

Solution: lint rule

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -150,13 +150,23 @@ module.exports = {
...
-        //     }
-        // ],
+        'no-restricted-imports': [
+            'error',
+            {
+                patterns: [
+                    {
+                        group: ['**/core/dist/*'],
+                        message:
+                            "Avoid importing from the core lib's dist/ folders, please use from or add to the exports instead.",
```
Comment (jpinkney-aws) [inline .eslintrc.js:160]: Is this supposed to say "please use from core directly" (or something like that?)

Comment (hayemaxi) [reply .eslintrc.js:160]: Hmm, how about
```suggestion
                            "Avoid importing from the core lib's dist/ folders; please use directly from the core lib exports.",
```

Comment (hayemaxi) [reply .eslintrc.js:160]: or
```suggestion
                            "Avoid importing from the core lib's dist/ folders; please use directly from the core lib defined exports.",
```

Comment (jpinkney-aws) [reply .eslintrc.js:160]: Either one LGTM

Comment (hayemaxi) [reply .eslintrc.js]: done

**Review by jpinkney-aws — APPROVED** [review]

**Comment by nkomonen-amazon** [comment]
nice

**Review by nkomonen-amazon — APPROVED** [review]

**Pull request merged**